agent/agent_endpoint_test.go

```diff
@@ -133,6 +133,482 @@ func TestAgent_Checks(t *testing.T) {
 	}
 }
 
+func TestAgent_Health_Service_Id(t *testing.T) {
```

Comment:
OPT/FAR: You should split the test in several smaller tests.
It is hard to remember which check was given to a service 100 lines before (same goes for service name test) and makes the test hard to read.